Product/Production/Gateway/DocumentDataSubmission_10/src/main/java/gov/hhs/fha/nhinc/docdatasubmission/v10/nhin/NhinDocDataSubmission.java

```diff
@@ -62,10 +59,12 @@ public InboundDocDataSubmission getInboundDocDataSubmission() {
     }
 
     @Override
+    @InboundMessageEvent(serviceType = "Document Data Submission", version = "1.0",
+        beforeBuilder = DocDataSubmissionBaseEventDescriptionBuilder.class,
+        afterReturningBuilder = DocDataSubmissionBaseEventDescriptionBuilder.class)
     public RegistryResponseType documentRegistryXDSRegisterDocumentSetB(RegisterDocumentSetRequestType body) {
-        LOG.debug("Calling service with context {} and inbound submission class {}", context,
-            inboundDocDataSubmission.getClass().getName());
-        return new RegistryResponseType();
+        return new NhinDocDataSubmissionImpl(inboundDocDataSubmission).documentRepositoryRegisterDocumentSetB(body,
```

Comment:
Optinal: Suggestion to add log to indicate Incomming nhin message from DDS so that we can troubleshoot in future